MoreLinq/MoreEnumerable.cs

```diff
@@ -35,5 +35,24 @@ public static partial class MoreEnumerable
                  : source is IReadOnlyCollection<T> readOnlyCollection ? readOnlyCollection.Count
                  : (int?)null;
         }
+
+        static int CountUpTo<T>(this IEnumerable<T> source, int max)
+        {
+            if (source == null) throw new ArgumentNullException(nameof(source));
+            if (max < 0) throw new ArgumentOutOfRangeException(nameof(max), "Max cannot be negative.");
```

Comment:
If `max` is allowed to be 0 then this method will produce an incorrect result due to the `++count == max` as the loop breaking condition. It can be made to work with a simpler and more robust while-loop:

```c#
while (count < max && e.MoveNext())
    count++;
```

The exception message should read as follows:

> The maximum count argument cannot be negative.

Because “Max” is not a proper word in English.